lift/src/main/scala/com/linkedin/lift/lib/testing/TestUtils.scala

```diff
@@ -52,4 +56,38 @@ object TestUtils {
       .config(sparkConf)
       .getOrCreate()
   }
+
+  /**
+    * Loading csv data
+    */
+  def loadCsvData(spark: SparkSession, dataPath: String, dataSchema: StructType, delimiter: String,
+    numPartitions: Int = 100): DataFrame ={
+    spark.read.format("csv")
+      .option("header", value = true)
+      .option("delimiter", delimiter)
+      .option("numPartitions", numPartitions)
+      .schema(dataSchema)
+      .load(dataPath)
+  }
+
+  case class DataWithPositionBias(itemId: Int, sessionId: Int, score: Double, position: Int, label: Int,
+    attribute: String)
+
+  /**
+    * To apply the effect of position bias (i.e. positive response decay), we multiply the labels with
+    * Bernoulli(1/(1 + position)) random numbers,
+    * where the position corresponds to the rank of an item in a session (according to item scores).
+    *
+    * @param spark sparksession
+    * @param dataWithoutPositionBias a dataset containing sessionId, score, position and label
+    * @return dataset with modified labels
+    */
+  def applyPositionBias(spark: SparkSession, dataWithoutPositionBias: Dataset[ScoreWithLabelAndAttribute]):
```

Comment:
We can obtain the `SparkSession` from the DataFrame or Dataset being passed. You can achieve this like so: `import dataWithoutPositionBias.sparkSesssion.implicits._`.

Reply:
thanks